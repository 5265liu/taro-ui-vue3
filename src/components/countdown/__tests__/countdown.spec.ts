import { mount } from '@vue/test-utils'
import AtCountdown from '../index'
import { sleep } from '@/tests/helper'

const factory = async (values = {}, slots = { default: [] }) => {
  const component = mount(AtCountdown as any, {
    slots,
    props: { ...values },
  })
  await sleep(0)
  return component
}

describe('AtCountdown Snap', () => {
  it('render initial AtCountdown', async () => {
    const wrapper = await factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props class', async () => {
    const wrapper = await factory({ class: 'test' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props style', async () => {
    const wrapper = await factory({ style: 'color:red;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props isCard', async () => {
    const wrapper = await factory({ isCard: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props format', async () => {
    const wrapper = await factory({ format: { hours: ':', minutes: ':', seconds: '' } })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props day', async () => {
    const wrapper = await factory({ day: 1 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props hours', async () => {
    const wrapper = await factory({ hours: 1 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props minutes', async () => {
    const wrapper = await factory({ minutes: 1 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props seconds', async () => {
    const wrapper = await factory({ seconds: 1 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- props isShowDay', async () => {
    const wrapper = await factory({ isShowDay: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- hours > 24 and show day', async () => {
    const wrapper = await factory({
      isShowDay: true,
      hours: 25,
    })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtCountdown -- hours > 24 and not show day', async () => {
    const wrapper = await factory({
      isShowDay: false,
      hours: 25,
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
