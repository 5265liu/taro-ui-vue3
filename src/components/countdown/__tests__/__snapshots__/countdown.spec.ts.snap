// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtCountdown Snap render AtCountdown -- hours > 24 and not show day 1`] = `
<view
  class="at-countdown"
>
  <!---->
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        25
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- hours > 24 and show day 1`] = `
<view
  class="at-countdown"
>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        01
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      天
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        01
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props class 1`] = `
<view
  class="test at-countdown test"
>
  <!---->
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props day 1`] = `
<view
  class="at-countdown"
>
  <!---->
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        24
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props format 1`] = `
<view
  class="at-countdown"
>
  <!---->
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      :
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      :
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    />
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props hours 1`] = `
<view
  class="at-countdown"
>
  <!---->
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        01
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props isCard 1`] = `
<view
  class="at-countdown at-countdown--card"
>
  <!---->
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props isShowDay 1`] = `
<view
  class="at-countdown"
>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      天
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props minutes 1`] = `
<view
  class="at-countdown"
>
  <!---->
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        01
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props seconds 1`] = `
<view
  class="at-countdown"
>
  <!---->
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        01
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render AtCountdown -- props style 1`] = `
<view
  class="at-countdown"
  style="color: red;"
>
  <!---->
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;

exports[`AtCountdown Snap render initial AtCountdown 1`] = `
<view
  class="at-countdown"
>
  <!---->
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      时
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      分
    </text>
  </view>
  <view
    class="at-countdown__item"
  >
    <view
      class="at-countdown__time-box"
    >
      <text
        class="at-countdown__time"
      >
        00
      </text>
    </view>
    <text
      class="at-countdown__separator"
    >
      秒
    </text>
  </view>
</view>
`;
